collector/loadavg_unix.go

```diff
@@ -39,32 +40,45 @@ func init() {
 // load1 stat.
 func NewLoadavgCollector() (Collector, error) {
 	return &loadavgCollector{
-		metric: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: Namespace,
-			Name:      "load1",
-			Help:      "1m load average.",
-		}),
+		metric: []prometheus.Gauge{
+			prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: Namespace,
+				Name:      "load1",
+				Help:      "1m load average.",
+			}),
+			prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: Namespace,
+				Name:      "load5",
+				Help:      "5m load average.",
+			}),
+			prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: Namespace,
+				Name:      "load15",
+				Help:      "15m load average.",
+			}),
+		},
 	}, nil
 }
 
 func (c *loadavgCollector) Update(ch chan<- prometheus.Metric) (err error) {
-	load, err := getLoad1()
+	loads, err := getLoad()
 	if err != nil {
 		return fmt.Errorf("couldn't get load: %s", err)
 	}
-	log.Debugf("Set node_load: %f", load)
-	c.metric.Set(load)
-	c.metric.Collect(ch)
+	for i, load := range loads {
+		log.Debugf("Set load %d: %f", i, load)
+		c.metric[i].Set(load)
+		c.metric[i].Collect(ch)
+	}
 	return err
 }
 
-func getLoad1() (float64, error) {
-	var loadavg [1]C.double
-	samples := C.getloadavg(&loadavg[0], 1)
+func getLoad() ([3]float64, error) {
```

Comment:
How about putting only the `getLoad()` function in separate, OS-specific files? Then the rest of the collector can be generic between Linux and other Unices. That way we don't have to maintain two sets of metrics names, etc.

It would just be `loadavg.go` then for the generic collector, and `loadavg_unix.go` / `loadavg_linux.go` for just the specific `getLoad()` functionality.